storage/badger/dkg_state.go

```diff
@@ -16,15 +18,34 @@ import (
 	"github.com/onflow/flow-go/storage/badger/transaction"
 )
 
-// DKGState stores state information about in-progress and completed DKGs, including
+// allowedStateTransitions defines the allowed state transitions for the Recoverable Random Beacon State Machine.
+var allowedStateTransitions = map[flow.DKGState][]flow.DKGState{
+	flow.DKGStateStarted:          {flow.DKGStateCompleted, flow.DKGStateFailure, flow.RandomBeaconKeyCommitted},
+	flow.DKGStateCompleted:        {flow.RandomBeaconKeyCommitted, flow.DKGStateFailure, flow.RandomBeaconKeyCommitted},
+	flow.RandomBeaconKeyCommitted: {flow.RandomBeaconKeyCommitted},
```

Comment:
I guess you meant "allow". It's because I have accounted for a possibility where in case of manual injection of key we might want to overwrite it again in case of mistake. But if we assume that any private key must comply with the `EpochCommit` event than this situation is just not possible in the first place. Idk, we can discuss this.